common/datavalues/src/types/deserializations/nullable.rs

```diff
@@ -46,6 +47,28 @@ impl TypeDeserializer for NullableDeserializer {
         Ok(())
     }
 
+    fn de_batch_with_nullable(
+        &mut self,
+        reader: &[u8],
+        step: usize,
+        rows: usize,
+        null_offset: usize,
+    ) -> Result<()> {
+        for row in 0..rows {
+            let mut reader = &reader[step * row..];
+            self.inner.de(&mut reader)?;
+            // null_offset -= self.inner().to_physical_type;
+            // todo, We should get the length of missed bytes which caused
+            // by self.inner.de method.
+            if reader[null_offset - 1] & 1 == 1 {
+                self.bitmap.push(false);
```

Comment:
self.bitmap.push(reader[null_offset - 1] & 1 != 1)

Reply:
> self.bitmap.push(reader[null_offset - 1] & 1 != 1)

That's cool.